### PR TITLE
feat: send web telemetry events to Datadog

### PR DESCRIPTION
**Pull request by macaptain**
## Overview

Using the telemetry_metric and telemetry_metrics_statsd reporter, we can send telemetry events to Datadog without defining more event handlers. The supervised process is configured with the same parameters as the Statix metrics generation.

In omg.status, we collect web events from Phoenix and send them to Datadog as metrics.

The fallback controller in WatcherRPC is also instrumented to execute telemetry events on errors, with route and error type recorded as tags.

image

## Changes

- Add dependencies :telemetry_metrics and :telemetry_metrics_statsd to omg.status
- Add a supervised task for handling :telemetry.execute calls and sending them to Datadog as metrics
- Send Phoenix router_dispatch metrics to Datadog
- Instrument the Fallback controller in WatcherRPC with telemetry for tracking error counts, routes and error types
- Assign error details to the `conn` in the Fallback controller
- Run 'mix format' across some files

## Testing

```
DD_API_KEY=<some-api-key> docker-compose -f docker-compose-watcher.yml -f docker-compose.dev.yml up
make docker-update-watcher
make docker-update-watcher_info
```

Make API calls to the Watcher endpoints, and check the local-development environment metrics explorer in Datadog. Look for metrics named "omg.watcher...".

### CODE REVIEW

File: apps/omg_status/lib/omg_status/application.ex [hunk 1]
```diff
@@ -39,22 +40,33 @@ defmodule OMG.Status.Application do
...
+              memsup_module: :memsup,
+              threshold: system_memory_high_threshold,
+              interval_ms: system_memory_check_interval_ms
+            ]
+          },
+          {
+            Telemetry,
+            [
+              release: Application.get_env(:omg_status, :release),
+              current_version: Application.get_env(:omg_status, :current_version)
```
Comment (InoMurko) [inline apps/omg_status/lib/omg_status/application.ex:56]: you can add these two into `OMG.Status.Configuration` and access them via this module

File: apps/omg_status/lib/omg_status/application.ex [hunk 2]
```diff
@@ -39,22 +40,33 @@ defmodule OMG.Status.Application do
...
-           [
-             dd_alarm_handler: OMG.Status.DatadogEvent.AlarmHandler,
-             release: Application.get_env(:omg_status, :release),
-             current_version: Application.get_env(:omg_status, :current_version),
-             publisher: OMG.Status.Metric.Datadog
-           ]}
+          {
+            AlarmConsumer,
+            [
+              dd_alarm_handler: OMG.Status.DatadogEvent.AlarmHandler,
```
Comment (InoMurko) [inline apps/omg_status/lib/omg_status/application.ex:64]:
```suggestion
              dd_alarm_handler: AlarmHandler,
```

File: apps/omg_status/lib/omg_status/application.ex [hunk 3]
```diff
@@ -39,22 +40,33 @@ defmodule OMG.Status.Application do
...
+              release: Application.get_env(:omg_status, :release),
+              current_version: Application.get_env(:omg_status, :current_version),
+              publisher: OMG.Status.Metric.Datadog
```
Comment (InoMurko) [inline apps/omg_status/lib/omg_status/application.ex:67]:
```suggestion
              publisher: Datadog
```

**Review by InoMurko — APPROVED** [review]
👏 👏 👏

File: apps/omg_status/lib/omg_status/application.ex (hunk 1 above)
Comment (macaptain) [reply apps/omg_status/lib/omg_status/application.ex]: ✅

**Review by T-Dnzt — APPROVED** [review]
🎉 🚀

**Pull request merged**